src/include/cell_inline.h

```diff
@@ -1272,12 +1272,21 @@ __wt_dsk_cell_data_ref(WT_SESSION_IMPL *session, int page_type, void *unpack_arg
 
 /*
  * __wt_page_cell_data_ref --
- *     Set a buffer to reference the data from an unpacked cell.
+ *     Set a buffer to reference the data from an unpacked cell. If we see an overflow removed cell,
```

Comment:
There are quite a few callers to this function - can they all handle a WT_RESTART error return in a sensible way? This is also performance critical code, so I wonder if we want to pay the cost of checking explicitly for overflow removed cells in all cases? My search shows:

```
src/reconcile/rec_visibility.c
178:        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, tmp));

src/txn/txn_rollback_to_stable.c
389:    WT_ERR(__wt_page_cell_data_ref(session, page, unpack, full_value));
757:            if ((ret = __wt_page_cell_data_ref(session, page, vpack, tmp)) == 0)

src/btree/bt_ret.c
236:        ret = __wt_page_cell_data_ref(session, page, &unpack, buf);
249:        ret = __wt_page_cell_data_ref(session, page, &unpack, buf);

src/btree/bt_page.c
260:            WT_ERR(__wt_page_cell_data_ref(session, page, &unpack, value));
304:            WT_ERR(__wt_page_cell_data_ref(session, page, &unpack, value));

src/btree/bt_debug.c
669:                          __wt_page_cell_data_ref(session, page, unpack, ds->t1));
```

Reply:
I have done the same search and decided that in the other places we should not return WT_RESTART. We can handle them as an assert.